import dlinfer before imageencoding

jinminxi104 opened the pull request:
## Motivation

vlm model inference causes cuda error on huawei device.

## Modification

import dlinfer before imageencoding(which causes cuda error)

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

irexyc commented:
The reason why initialize ImageEncoder before llm is that the llm model may be large with pre-allocated kvcache. If initialize llm first, the left cuda memory may be insufficient to load the vision model.

jinminxi104 commented:
> The reason why initialize ImageEncoder before llm is that the llm model may be large with pre-allocated kvcache. If initialize llm first, the left cuda memory may be insufficient to load the vision model.

I reverted the order of initialize ImageEncoder and llm; and added try_import_deeplink.

irexyc commented:
Please use `pre-commit` to fix the lint error.
